clusterctl/examples/google/generate-yaml.sh

```diff
@@ -36,9 +67,85 @@ while test $# -gt 0; do
         esac
 done
 
-if [ $OVERWRITE -ne 1 ] && [ -f $GENERATED_FILE ]; then
-  echo File $GENERATED_FILE already exists. Delete it manually before running this script.
+if [ $OVERWRITE -ne 1 ] && [ -f $MACHINE_GENERATED_FILE ]; then
+  echo File $PROVIDERCOMPONENT_GENERATED_FILE already exists. Delete it manually before running this script.
+  exit 1
+fi
+
+if [ $OVERWRITE -ne 1 ] && [ -f $CLUSTER_GENERATED_FILE ]; then
+  echo File $PROVIDERCOMPONENT_GENERATED_FILE already exists. Delete it manually before running this script.
+  exit 1
+fi
+
+if [ $OVERWRITE -ne 1 ] && [ -f $PROVIDERCOMPONENT_GENERATED_FILE ]; then
+  echo File $PROVIDERCOMPONENT_GENERATED_FILE already exists. Delete it manually before running this script.
+  exit 1
+fi
+
+if [ $OVERWRITE -ne 1 ] && [ -f $ADDON_GENERATED_FILE ]; then
+  echo File $ADDON_GENERATED_FILE already exists. Delete it manually before running this script.
   exit 1
 fi
 
-sed -e "s/\$GCLOUD_PROJECT/$GCLOUD_PROJECT/" $TEMPLATE_FILE | sed -e "s/\$ZONE/$ZONE/" > $GENERATED_FILE
+if [ ! -f $MACHINE_CONTROLLER_SA_FILE ]; then
+  echo Generating $MACHINE_CONTROLLER_SA_EMAIL service account for machine controller
```

Comment:
Long term, I see this going in two directions:
(1) A bunch of these service accounts will be handled by the cluster controller. Ideally, the only account we need to create is one for the cluster controller itself.
(2) This script will likely turn into an executable in the future if it remains complex. Once it is an executable, it can re-use libraries.

Reply:
ahh ok that makes a lot of sense. I think number 1 makes sense short term.